meshtastic/tests/test_node_runtime_content.py

```
@@ -0,0 +1,1118 @@
+    def test_read_ringtone_ignores_late_callback_after_timeout(
+        self,
+        mock_node_for_admin: MagicMock,
+        cache_store: _NodeContentCacheStore,
+        response_runtime: _NodeContentResponseRuntime,
+    ) -> None:
+        """Late callbacks from timed-out reads should not contaminate subsequent reads."""
+        callback_by_call: dict[int, Any] = {}
+        send_call_count = {"count": 0}
+
+        def send_admin_side_effect(
+            _message: admin_pb2.AdminMessage, **kwargs: Any
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)